### ISSUE TITLE
Lack of mqtt message for TEXT devices

### ISSUE THREAD

**Issue by AdamWeglarz** [domoticz/domoticz#6135]
Hello,

Issue is simple to reproduce.
Create text device. In my case it is Satel alarm default device Output:zanik230V (lack of 230v supply message).
Add MQTT Client gateway with LAN Interface (I have publish topic Index with retain)
This device does not report its state via MQTT, when other devices do.
I am on domoticz 
Version: 2024.7
Build Hash: b317cfab1
Compile Date: 2024-07-13 16:59:31

**Comment by gizmocuz** [comment]
@AdamWeglarz , instead of a text message, this could also be a switch (on/off = yes/no), you can edit the switch to set it to another type (For instance Alarm)

What/who is updating the text sensor? Is this the Satel hardware or a script?
If it is the Satel Hardware, I think this happens in 

**SatelIntegra::ReportOutputState**

**m_sql.UpdateValue(m_HwdID, 0, sTmp.c_str(), 1, pTypeGeneral, sTypeTextStatus, 12, 255, 0, state ? "On" : "Off", devname, true, m_Name.c_str());**

Not sure why this was not programmed as a switch

Are you compiling from source, or can you compile/test from source?

If you change that line to:

		**SendGeneralSwitch(Idx, 3, 255, state ? gswitch_sOn : gswitch_sOff, 0, "", m_Name);**

Then this could be solved

@szczukot, could this be changed to a switch? Why did you choose for a Text sensor (with On/Off as value), if it was a switch, you could also set an alarm/notification on this in the system (and use it in the eventsystem)

**Comment by Stolowka** [comment]
Yes, I can make the change, but which subtype should I choose? This switch needs to be read-only.

**Comment by szczukot** [comment]
Yes, I can make the change, but which subtype should I choose? This switch needs to be read-only.

**Comment by gizmocuz** [comment]
Contact type? But the user can do this
You can also add to the name (read only)

**Comment by szczukot** [comment]
Changing by the user makes no sense. These are readonly outputs, so it doesn't make sense if someone clicked them in Domoticz. It's better to set some read-only type right away

**Comment by gizmocuz** [comment]
It's very easy to select a type by the user, but by all means do make it the way you would like to make it 👍👌
